### PR TITLE
chore(next-release/main): Callout, YoutubeEmbed, minor content format updates

### PR DESCRIPTION
**Pull request by hbuchel**
#### Description of changes:

Updates to Callout and YoutubeEmbed; some minor updates to content formatting.

https://next-release-componentupdates.d2bfwhpcsj9awv.amplifyapp.com/javascript/test-page/

### Instructions

**If this PR should not be merged upon approval for any reason, please submit as a DRAFT**

Which product(s) are affected by this PR (if applicable)?
- [ ] amplify-cli
- [ ] amplify-ui
- [ ] amplify-studio
- [ ] amplify-hosting
- [ ] amplify-libraries

Which platform(s) are affected by this PR (if applicable)?
- [ ] JS
- [ ] iOS
- [ ] Android
- [ ] Flutter
- [ ] React Native

**Please add the product(s)/platform(s) affected to the PR title**

#### Checks

- [ ] Does this PR conform to [the styleguide](https://github.com/aws-amplify/docs/blob/main/STYLEGUIDE.md)?

- [ ] Does this PR include filetypes other than markdown or images? Please add or update unit tests accordingly.

- [ ] Are any files being deleted with this PR? If so, have the needed redirects been created?

- [ ] Are all links in MDX files using the MDX link syntax rather than HTML link syntax? <br /> 
      _ref: MDX: `[link](https://link.com)` 
            HTML: `<a href="https://link.com">link</a>`_

### When this PR is ready to merge, please check the box below
- [ ] Ready to merge

_By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license._

### CODE REVIEW

File: mdx-components.tsx
Comment (hbuchel) [inline mdx-components.tsx:1]: I cleaned up some components that I don't *think* we'll expect authors to need/use.

File: src/components/Callout/Callout.tsx
Comment (hbuchel) [inline src/components/Callout/Callout.tsx:1]: The props for this are a little weird, but I kept them as is so there is less to update when we pull in the existing content. Callout now uses `<Message />` as the base, and outputs the info/warning color themes.

File: src/components/Menu/Menu.tsx
Comment (hbuchel) [inline src/components/Menu/Menu.tsx:1]: We missed some missing keys in the nav menu PR so added them here.

File: src/pages/index.tsx [hunk 1]
```diff
@@ -46,7 +46,7 @@ export default function Page() {
           libraries, CLI, and services, you can easily connect your frontend to
           the cloud for authentication, storage, APIs, and more.
         </Text>
-        <Flex>
+        <Flex className="home-intro__cta">
```
Comment (hbuchel) [inline src/pages/index.tsx:49]: This class makes the buttons not be all weird and tall on mobile.

File: src/themes/defaultIcons.tsx
Comment (hbuchel) [inline src/themes/defaultIcons.tsx:1]: These are for the [IconProvider](https://ui.docs.amplify.aws/react/components/message#icons)

**Review by hbuchel — COMMENTED** [review]
Some callouts for changes:

**Review by timngyn — APPROVED** [review]

**Review by sowmyadev-kota — APPROVED** [review]

**Review by katiegoines — APPROVED** [review]

**Pull request merged**